### PR TITLE
Update instructions in documentation for OAuth examples

### PR DESCRIPTION
**Pull request by zimeg**
### Summary

This PR makes a few changes to the OAuth examples in documentation to remove some confusion:

- Specifies that the `/slack/oauth` is added to the server URL for `*_PATH` environment variables to avoid unknowingly creating paths like `/slack/oauth/slack/oauth/start`.
- Updates the map entries when creating a server to make the map mutable to avoid a compilation error.

### Category

* [x] **bolt** (Bolt for Java)
* [ ] **bolt-{sub modules}** (Bolt for Java - optional modules)
* [ ] **slack-api-client** (Slack API Clients)
* [ ] **slack-api-model** (Slack API Data Models)
* [ ] **slack-api-*-kotlin-extension** (Kotlin Extensions for Slack API Clients)
* [ ] **slack-app-backend** (The primitive layer of Bolt for Java)
* [x] **docs** (Documentation)

## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/java-slack-sdk/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you agree to those rules.

### CODE REVIEW

File: docs/guides/app-distribution.md [hunk 1]
```diff
@@ -59,10 +60,10 @@ apiApp.command("/hi", (req, ctx) -> {
 App oauthApp = new App().asOAuthApp(true);
 
 // Mount the two apps with their root path
-SlackAppServer server = new SlackAppServer(Map.of(
+SlackAppServer server = new SlackAppServer(new HashMap<>(Map.ofEntries(
   entry("/slack/events", apiApp), // POST /slack/events (incoming API requests from the Slack Platform)
   entry("/slack/oauth", oauthApp) // GET  /slack/oauth/start, /slack/oauth/callback (user access)
-));
```
Comment (zimeg) [inline docs/guides/app-distribution.md:65]: This raises the following error for me on openJDK 19:

```
[main] INFO org.eclipse.jetty.util.log - Logging initialized @2068ms to org.eclipse.jetty.util.log.Slf4jLog
Exception in thread "main" java.lang.UnsupportedOperationException
        at java.base/java.util.ImmutableCollections.uoe(ImmutableCollections.java:142)
        at java.base/java.util.ImmutableCollections$AbstractImmutableMap.putAll(ImmutableCollections.java:1080)
        at com.slack.api.bolt.jetty.SlackAppServer.<init>(SlackAppServer.java:121)
        at com.slack.api.bolt.jetty.SlackAppServer.<init>(SlackAppServer.java:70)
        at NitroApp.main(ExampleApp.java:130)
```

File: docs/guides/app-distribution.md [hunk 2]
```diff
@@ -78,11 +79,11 @@ Here is the list of the necessary configurations for distributing apps built wit
...
-|**SLACK_REDIRECT_URI_PATH**|**Path for OAuth Redirect URI**: This endpoint handles callback requests after the Slack's OAuth confirmation. The path must be consistent with **SLACK_REDIRECT_URI** value. The suggested path is `/slack/oauth/callback` but you can go with any path.|
+|**SLACK_INSTALL_PATH**|**Starting point of OAuth flow**: This endpoint redirects users to the Slack Authorize endpoint with required query parameters such as `client_id`, `scope`, `user_scope` (only for v2), and `state`.<br><br>The suggested path is `/slack/oauth/start` but you can go with any path. Note that the example above automatically prepends `/slack/oauth` to this variable.|
+|**SLACK_REDIRECT_URI_PATH**|**Path for OAuth Redirect URI**: This endpoint handles callback requests after the Slack's OAuth confirmation. The path must be consistent with **SLACK_REDIRECT_URI** value.<br><br>The suggested path is `/slack/oauth/callback` but you can go with any path. Note that the example above automatically prepends `/slack/oauth` to this variable.|
```
Comment (zimeg) [inline docs/guides/app-distribution.md:86]: This displays like so:

docs

**Review by zimeg — COMMENTED** [review]
📝 Some notes for the reviewers!

**Review by filmaj — COMMENTED** [review]
I'm not really sure the implication of changing the mapping class used here and how that affects which Java runtime will have compatibility (or not), so I would prefer to wait for Kaz to provide his opinion.

**Review by seratch — APPROVED** [review]
Thank you!

**Pull request merged**